streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java

```diff
@@ -103,8 +104,12 @@ public void init(final ProcessorContext context,
 
     @SuppressWarnings("unchecked")
     void initStoreSerde(final ProcessorContext context) {
+        final InternalProcessorContext internalProcessorContext = (InternalProcessorContext) context;
+        final String storeName = name();
+        final String changelogTopic = internalProcessorContext.changelogFor(storeName);
         serdes = new StateSerdes<>(
-            ProcessorStateManager.storeChangelogTopic(context.applicationId(), name()),
+            changelogTopic != null
+                ? changelogTopic : ProcessorStateManager.storeChangelogTopic(context.applicationId(), storeName),
```

Comment:
nit: move `?` to the line above and move `ProcessorStateManager.storeChangelogTopic(context.applicationId(), storeName),` to its own line.

Preferred formatting (makes it easy to read):
```
condition ?
  then :
  else
```

Reply:
Ack